### ISSUE TITLE
add the possibility to create custom ports

### ISSUE THREAD

**Issue by jsen-** [kubernetes-sigs/cluster-api-provider-openstack#788]
/kind feature

**Describe the solution you'd like**
Some of the workloads in our clusters require network acceleration (SR-IOV). In order to support this requirement, we'd like to be able to create and attach custom ports to worker nodes.

We tried attaching the ports to compute instances after they were created, unfortunately this does not work for some reason, the VM needs to be created with the ports already attached.

We were also playing with a possibility to pre-create the ports and only let `capo` attach them during VM creation, but this does not fit the the ClusterAPI design - we only have access to `OpenstackMachineTemplate` and one port can be only attached to one instance.

It seems most natural to let `capo` also create the ports based on a [user-defined template](https://github.com/gophercloud/gophercloud/blob/bcd179f2c98de61525a99bf9310d698fd7786a17/openstack/networking/v2/ports/requests.go#L112-L125).
The most important settings for us is actually [vnic_type](https://github.com/gophercloud/gophercloud/blob/bcd179f2c98de61525a99bf9310d698fd7786a17/openstack/networking/v2/extensions/portsbinding/requests.go#L18) which needs to be set to direct.

**Comment by sbueringer** [comment]
@jsen- Overall this seems reasonable to me. 

Do you need to create additional ports or is it sufficient to customize the one port we create anyway (via CreateOpts)?

**Comment by jichenjc** [comment]
from https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/778/files#diff-5e7dbfd52305a21a11fb94fab201062ae2c7b851db93c2a8443a1d82d7aa7936R253
 seems a couple of ports will be created ,that's allowed and from this issue seems vnic_type is the key thing to add

**Comment by macaptain** [comment]
This is a feature I'd be really interested in having too. In particular, I'd like to have machines with multiple ports on different networks and to be able to configure the settings per port. `vnicType` is also something I'd like to use. It's great to see there's already an issue for this feature and work in progress on it.

@jsen- , I built your branch implementing this feature and I used it to create a custom port defined in the `OpenStackMachineTemplate`. This worked nicely for me, so thank you for your work. I noticed the pull request hadn't received any updates for about a month though. Is this something you are still looking at and working on? I'm new to the CAPO project but I'd be happy to help if I can.
